src/main/java/apoc/coll/Coll.java

```diff
@@ -874,4 +875,33 @@ public boolean different(@Name("values") List<Object> values) {
 
         return newList;
     }
+
+    @UserFunction
+    @Description("apoc.coll.extract(coll, property) - extracts the property from each element in the coll and returns a list of the values ")
+    public List<Object> extract(@Name("coll") List<Object> coll, @Name( "property" ) String property) {
+        if (coll == null || coll.size() == 0) {
+            return Collections.emptyList();
+        }
+        List<Object> extraction = new ArrayList<>(  );
+        Object o = coll.get( 0 );
```

Comment:
We should check it per instance in the loop, and possibly add a null if it's neither map nor graph entity.